pkg/webhook/server.go

```diff
@@ -103,3 +111,40 @@ func (s *Server) setCert(name, namespace string) error {
 
 	return s.onCertChange(webhookCert)
 }
+
+func (s *Server) certCommonName() string {
+	return fmt.Sprintf("%s.%s.svc", s.name, s.namespace)
+}
+
+// configureCertificateRotation gets the certificate from the certificate manager and spawns a goroutine to watch for certificate rotation.
+func (s *Server) configureCertificateRotation() func() {
```

Comment:
This is using the same pattern in https://github.com/openservicemesh/osm/blob/7abf10d787d37cbc79c207888e70248c3a9eef66/pkg/ingress/gateway.go#L116. 

There might be opportunity to refactor and re-use.